src/rgw/rgw_op.cc

```diff
@@ -6359,9 +6364,103 @@ void RGWCompleteMultipart::pre_exec()
   rgw_bucket_object_pre_exec(s);
 }
 
+static inline int
+try_sum_part_cksums(const DoutPrefixProvider *dpp,
+		    CephContext *cct,
+		    rgw::sal::MultipartUpload* upload,
+		    RGWMultiCompleteUpload* parts,
+		    std::optional<int>& num_parts,
+		    std::optional<rgw::cksum::Cksum>& out_cksum,
+		    optional_yield y)
+{
+  /* 1. need checksum-algorithm header (if invalid, fail)
+     2. conditional on have-checksum,
+     3. need digest for supplied algo
+     4. iterate over parts, confirm each has same algo, if not, fail
+     5. for each part-checksum, accumlate bytes into new checksum
+     6. return armored and append "-<nparts>"
+     7. verify -- if invalid, fail */
+
+    /* rgw_sal.h says that list_parts is called for the side effect of loading
+     * the parts of an upload into "cache"--the api is strange and truncated
```

Comment:
The "cache" is just a map of MultipartPart objects.  Resuming a truncated list_parts() call on the same MultipartUpload object continues to fill it's map, so a complete list_parts() run (i.e. until !truncated) should have a complete map of parts.

Reply:
but if we hit the truncated case, and then re-enter list_parts(), what happens to the cached info elements?